x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/alerts/helpers/alerting_api_helper.ts

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client, errors } from '@elastic/elasticsearch';
+import { ParsedTechnicalFields } from '@kbn/rule-registry-plugin/common';
+import pRetry from 'p-retry';
+import { ApmRuleType } from '@kbn/rule-data-utils';
+import { ApmRuleParamsType } from '@kbn/apm-plugin/common/rules/schema';
+import { ApmDocumentType } from '@kbn/apm-plugin/common/document_type';
+import { RollupInterval } from '@kbn/apm-plugin/common/rollup';
+import { ObservabilityApmAlert } from '@kbn/alerts-as-data-utils';
+import type { ApmApiClient } from '../../../../../services/apm_api';
+import type { SupertestWithRoleScopeType } from '../../../../../services';
+
+export const APM_ALERTS_INDEX = '.alerts-observability.apm.alerts-*';
+export const APM_ACTION_VARIABLE_INDEX = 'apm-index-connector-test';
+
```

Comment:
I moved a few things there but we might want to review the functions in the `AlertingApiProvider`. There is a lot of duplication in the `helper` object. Few things from `helper` could be extracted out a returned in the main object. I could do that, but it would extrapolate the scope of this PR

Reply:
Totally agree.  
When that work starts it will be the third or fourth step of bringing the tech debt of way too many alerting functions down.  It's been an on-going process.  Hence, why I was a little alarmed about yet another alerting helper file.

Reply:
Yeah, I think these common API's could also use some documentation too. I was unaware that I could use it outside alerting tests context.

Reply:
Indeed.  I'm creating a ticket for this now, I'll add something about the documentation too

Reply:
[Issue](https://github.com/elastic/kibana/issues/199619) to track the documentation ask and the duplication problem.